build.gradle.kts

```diff
@@ -15,4 +15,14 @@ subprojects {
       targetCompatibility = JavaVersion.VERSION_11
     }
   }
+
+  pluginManager.withPlugin("java-library") {
+
+    dependencies {
+      "api"(platform(project(":hypertrace-core-graphql-platform")))
```

Comment:
just curious. Why is this `"api"` and not just `api`?

Reply:
that's a quirk of the kotlin compiler here - normally we add plugins statically, which are pre-processed and add certain extensions to the scope (such as the api configuration). Because I'm doing this in a subprojects block, kotlin doesn't know what plugins are available for the specific plugin it's iterating through so the compiler won't let me use anything plugin-specific. I guard against this on line 19, but it means I need to access this in an unsafe way. Alternately, I could have done something like `val api = configurations.getByName("api")`, but that felt more cumbersome and is just as unsafe.